libbeat/common/docker/watcher_test.go

```diff
@@ -472,16 +486,37 @@ func runWatcherShortID(t *testing.T, kill bool, containers [][]types.Container,
 		t.Fatal("'watcher' was supposed to be pointer to the watcher structure")
 	}
 
-	err = watcher.Start()
-	if err != nil {
-		t.Fatal(err)
-	}
+	return watcher, client.done
+}
 
-	<-client.done
-	if kill {
-		watcher.Stop()
-		watcher.stopped.Wait()
-	}
+func runAndWait(w *watcher, done chan interface{}) *watcher {
```

Comment:
Just wondering, do we actually pass this channel in any of the calls?

Reply:
`runAndWait` is used with the returned values of `testWatcher`/`testWatcherShortID`, they return a watcher and a done channel, exactly the parameters `runAndWait` expect.
A `nil` channel would block.

Reply:
👍 thanks for clarifying